torchgen/dest/lazy_ir.py

```diff
@@ -290,9 +286,12 @@ def gen(self, schema: LazyIrSchema) -> List[str]:
         members_to_string = []
         for arg in scalar_args:
             if isinstance(arg.lazy_type, OptionalCType):
+                value = f"{arg.name}.value()"
+                if arg.is_generator:
+                    value = '"torch.Generator()"'
```

Comment:
can you refresh me on how this works in generated code?

is a Generator object going to be an 'input' to the graph, or a part of the graph?

if it's part of the graph then we should be sure that the lazy graph is recompiling every time you use a new generator object.  Is that the way it works?  Would be good to expand the tests to demonstrate this

Reply:
> is a Generator object going to be an 'input' to the graph, or a part of the graph?

It is a part of the graph.

> if it's part of the graph then we should be sure that the lazy graph is recompiling every time you use a new generator object. Is that the way it works?

Yes, the generator state is included in the [hash](https://github.com/antoniojkim/pytorch/blob/antoniojkim/torchscript_generator_support/torch/csrc/lazy/core/hash.h#L152-L154). So, if a different generator with a different state is used, it will recompile.

Recompilation is most likely fine in this case as we don't anticipate users using anything but the default generator inside a training/inference loop. We only anticipate them using it during initialization for deterministic parallel weight initialization.

> Would be good to expand the tests to demonstrate this

Yep, will do

Reply:
A test was added to verify this behaviour